File: data structures and algorithms/python/04_017.py
from abc import ABC, abstractmethod

class Observer(ABC):
    @abstractmethod
    def update(self, message):
        pass

class NewsAgency:
    def __init__(self):
        self.observers = []
    
    def add_observer(self, observer):
        self.observers.append(observer)
    
    def notify_observers(self, news):
        for observer in self.observers:
            observer.update(news)

File: data structures and algorithms/cs/04_017.cs
using System.Collections.Generic;

public interface IObserver
{
    void Update(string message);
}

public class NewsAgency
{
    private List<IObserver> observers = new List<IObserver>();

    public void AddObserver(IObserver observer)
    {
        observers.Add(observer);
    }

    public void NotifyObservers(string news)
    {
        foreach (IObserver observer in observers)
        {
            observer.Update(news);
        }
    }
}
